src/execution/execute.ts

```diff
@@ -504,10 +514,37 @@ export function buildExecutionContext(
     typeResolver: typeResolver ?? defaultTypeResolver,
     subscribeFieldResolver: subscribeFieldResolver ?? defaultFieldResolver,
     subsequentPayloads: new Set(),
+    executionController: new ExecutionController(signal),
     errors: [],
   };
 }
 
+class ExecutionController {
+  /** For performance reason we can't use `signal.isAborted` so we cache it here */
+  isAborted: boolean = false;
```

Comment:
For more context, what are the performance implications you mention here?

Reply:
I will add a screenshot tomorrow.
But you can run `npm run benchmark`(locally) on top of my commit and see the difference.